spec/requests/list_controller_spec.rb

```diff
@@ -759,6 +759,37 @@
       get "/c/hello/world/bye/#{subsubcategory.id}"
       expect(response.status).to eq(301)
       expect(response).to redirect_to("/c/#{category.slug}/#{subcategory.slug}/#{subsubcategory.slug}/#{subsubcategory.id}")
+
+      get "/c/#{category.slug}/#{subcategory.slug}/#{subsubcategory.slug}/#{subsubcategory.id}"
+      expect(response.status).to eq(200)
+    end
+
+    it "redirects to URL with correct case slug" do
+      category.update(slug: "hello")
+
+      get "/c/Hello/#{category.id}"
+      expect(response).to redirect_to("/c/hello/#{category.id}")
+
+      get "/c/hello/#{category.id}"
+      expect(response.status).to eq(200)
+    end
+
+    context "does not create a redirect loop" do
+      it "with encoded slugs" do
+        category = Fabricate(:category)
+        category.update_columns(slug: CGI.escape("systèmes"))
```

Comment:
```suggestion
        category = Fabricate(:category, slug: CGI.escape("systèmes"))
```

Will this work so that we can fabricate the record in a single query?